src/classes/PMT_Payment_TEST.cls

```diff
@@ -257,6 +257,188 @@ private class PMT_Payment_TEST {
+        try {
+            insert invalidPaidPayment;
+        } catch (Exception e) {
+            System.assert(e.getMessage().contains('A paid payment must have a payment date.'), 'The error message is not expected, actual: ' + e.getMessage());
+        }
+
+        // Test the invalid unpaid insertion case
+        npe01__OppPayment__c invalidUnpaidPayment = new npe01__oppPayment__c(npe01__Opportunity__c = opp.Id, npe01__Paid__c = false);
+
+        try {
+            insert invalidUnpaidPayment;
+        } catch (Exception e) {
+            System.assert(e.getMessage().contains('An unpaid payment must have a scheduled date.'), 'The error message is not expected, actual: ' + e.getMessage());
+        }
+
+        // Test the valid paid paymennt insertion case
+        npe01__OppPayment__c validPaidPayment = new npe01__oppPayment__c(npe01__Opportunity__c = opp.Id, npe01__Paid__c = true, npe01__Payment_Date__c = Date.today());
+        insert validPaidPayment;
+
+        validPaidPayment = queryForPaymentForTesting(validPaidPayment.Id);
+
+        System.assertEquals(true, validPaidPayment.npe01__Paid__c, 'Paid should be true, but it is actually: ' + validPaidPayment.npe01__Paid__c);
+        System.assertEquals(Date.Today(), validPaidPayment.npe01__Payment_Date__c, 'The payment date should be today, but it is actually: ' + validPaidPayment.npe01__Payment_Date__c);
+
+        // Test the valid unpaid payment insertion case
+        npe01__OppPayment__c validUnpaidPayment = new npe01__oppPayment__c(npe01__Opportunity__c = opp.Id, npe01__Paid__c = false, npe01__Scheduled_Date__c = Date.today());
+        insert validUnpaidPayment;
+
+        validUnpaidPayment = queryForPaymentForTesting(validUnpaidPayment.Id);
+
+        System.assertEquals(false, validUnpaidPayment.npe01__Paid__c, 'Paid should be false, but it is actually: ' + validUnpaidPayment.npe01__Paid__c);
+        System.assertEquals(Date.Today(), validUnpaidPayment.npe01__Scheduled_Date__c, 'The payment date should be today, but it is actually: ' + validUnpaidPayment.npe01__Scheduled_Date__c);
+
+        Integer validPaymentsCount = [Select Count() from npe01__OppPayment__c Where Id =: validPaidPayment.Id OR Id =: validUnpaidPayment.Id];
+        System.assertEquals(2, validPaymentsCount, 'There is an incorrect number of payments');
+
+        // *************************** Test Update *******************************
+
+        npe01__OppPayment__c pmt = new npe01__oppPayment__c(npe01__Opportunity__c = opp.Id, npe01__Paid__c = true, npe01__Payment_Date__c = Date.Today());
+        insert pmt;
+        // Test the invalid paid payment update case 
+        pmt = updatePaymentForTesting(pmt, true, false, false);
+
+        try {
+            update pmt;
+        } catch (Exception e) {
+            System.assert(e.getMessage().contains('A paid payment must have a payment date.'), 'The error message is not expected, actual: ' + e.getMessage());
```

Comment:
When you change to Custom Labels, update here too.

Reply:
Updated.